xtts_api_server/server.py

```
@@ -1,345 +1,377 @@
+    # An attacker may summon this endpoint with ../../etc/passwd and recover the password file of your PC (in linux) or access any other file on the PC
+    if ".." in file_name:
+        raise HTTPException(status_code=404, detail=".. in the file name! Are you kidding me?")        
+    # Robust path traversal protection
+    try:
+        base = Path(XTTS.speaker_folder).resolve(strict=True)
+        target = (base / file_name).resolve()
+        # Ensure the resolved target is under the base directory
+        if not str(target).startswith(str(base) + os.sep):
+            raise HTTPException(status_code=404, detail="Invalid file path")
+    except Exception:
+        raise HTTPException(status_code=404, detail="Invalid file path")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)